osmtm/views/user.py

```diff
@@ -44,10 +44,15 @@ def users(request):
 def users_json(request):
     query = DBSession.query(User).order_by(User.username)
 
+    q = ''
     if 'q' in request.params:
-        q = request.params.get('q')
-        query = query.filter(User.username.ilike('%' + q + '%')).limit(10)
+        q = request.params.get('q').strip()
 
+    if q:
+        query = query.filter(User.username.ilike('%' + q + '%')).limit(10)
+    else:
+        request.response.cache_control = \
+            'private, max-age=600'
```

Comment:
It would be interesting to look at other server side caching - this could presumably be generated once every 10 minutes by cron and rendered as a static asset, with appropriate etags and other headers - would reduce DB load even further.